main.go

```diff
@@ -126,16 +126,39 @@ func executeCreateRelease(ghActionIface utils.GithubActionIface, githubSHA, next
 	return nil
 }
 
+func isSkipReleaseLabelFound(ghActionIface utils.GithubActionIface, eventPath string) (bool, error) {
+	for _, labelName := range []string{"skip-release", "skipRelease"} {
+		isSkipRelease, err := ghActionIface.DoesLabelExist(labelName, eventPath)
+		if err != nil {
+			return false, err
+		}
+		if isSkipRelease {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// TODO: ./main.go:157:10: elseif: can replace 'else {if cond {}}' with 'else if cond {}'
 func executeAction(ghActionIface utils.GithubActionIface, actionConfig ActionConfig) {
+	isSkipRelease, err := isSkipReleaseLabelFound(ghActionIface, actionConfig.EventPath)
+	if err != nil {
+		Exit(1)
+	}
 	core.Info("Executing PR guard now")
 	// This will prevent the action from running if the guard fails
-	err := executeGuard(ghActionIface, actionConfig.ReleaseBranch, actionConfig.EventPath)
+	err = executeGuard(ghActionIface, actionConfig.ReleaseBranch, actionConfig.EventPath)
 	if err != nil {
 		core.Error(err.Error())
-		if err == ErrPRNotBase || err == ErrEmptyOption || err == ErrNoValidSemVerLabelFound {
+		if isSkipRelease && err == ErrNoValidSemVerLabelFound {
 			Exit(1)
 		}
-		Exit(0)
+		if !isSkipRelease {
+			if err == ErrPRNotBase || err == ErrEmptyOption || err == ErrNoValidSemVerLabelFound {
+				Exit(1)
+			}
+			Exit(0)
+		}
```

Comment:
Not sure about that, but I think you still need some error handling in a 'simple' line - when there is an error, but it won't enter any of the 'ifs'

Reply:
We want to do Exit(0) most of the times when not skip release. When skip release is enabled then we don't want to Exit(0) - we only want to Exit(1) on actual errors